Market update 11

limichange opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Introduced tabbed navigation for market views, allowing users to easily switch between Watchlist and Trending tokens on both desktop and mobile.
  * Added a modal route for detailed market views, enabling smoother navigation and improved user experience.
  * Enhanced scroll behavior in transaction history, preventing unwanted scroll jumps when new data loads.

* **Improvements**
  * Updated styling and layout for holders and network selector components for better visual consistency.
  * Increased the number of market token pages fetched, providing more data for users on all platforms.
  * Simplified and unified filter bar and watchlist handling for a cleaner interface.

* **Removals**
  * Removed the custom MarketViewToggle component and related props, consolidating view switching into the new tabbed interface.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
refactor: update SwapPanel to utilize InModalDialog for swap actions, enhancing user experience with improved modal presentation

limichange committed:
fix: update default TradingView URL for local development

limichange committed:
Merge branch 'x' into market-update-11

revan-zhang commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/04173315-0c07-408e-896c-f532ddcdd26a?refs=5d54a24149df6c61300b56160a9b5c7883e0315d&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/04173315-0c07-408e-896c-f532ddcdd26a/license?refs=5d54a24149df6c61300b56160a9b5c7883e0315d&source=prChecksComment)

limichange committed:
refactor: clean up MarketHomeV2 components by removing unused atoms and props

- Removed showWatchlistOnlyAtom from atoms.ts as it was no longer needed.
- Updated MarketHomeV2 to eliminate the use of recommendedTokens and activeTab state, simplifying the component logic.
- Refactored MarketFilterBar and MarketTokenList components to remove dependencies on showWatchlistOnly.
- Enhanced layout components to utilize Tabs for better organization of watchlist and trending views.
- Improved code readability and maintainability by standardizing prop usage across components.

limichange committed:
refactor: enhance DesktopLayout in MarketHomeV2 for improved layout and organization

- Updated Stack components to ensure full height usage for better layout consistency.
- Wrapped the watchlist tab content in a Tabs.ScrollView for improved scrolling behavior.
- Adjusted padding properties for better spacing and visual alignment across components.

limichange committed:
feat: enhance MarketDetailV2 and MarketHomeV2 with new routing and layout improvements

- Added conditional header visibility for MarketDetailV2 based on platform environment.
- Introduced showWatchlistOnlyAtom to manage watchlist visibility state.
- Updated navigation logic in useToDetailPage to support modal presentation on native platforms.
- Adjusted layout properties for HolderItemNormal to improve UI consistency.
- Added placeholder implementation for MarketViewToggle component.
- Updated routing configuration to include MarketDetailV2 modal for better navigation experience.
- Fixed default TradingView URL for production use.

limichange committed:
feat: enhance TransactionsHistory component with scroll handling and auto-scroll functionality

- Added useEffect to scroll to the top of the TransactionsHistory list when transactions update, provided the user hasn't scrolled.
- Implemented handleScroll function to track user scroll behavior and manage scroll state.
- Updated ListView to utilize a ref for scrolling and added onScroll event handling for improved user experience.

limichange committed:
fix: adjust scroll threshold in TransactionsHistory and update page count in useMarketTokenList

- Reduced the scroll distance threshold for marking user scroll state in TransactionsHistory from 20 to 10 for improved responsiveness.
- Updated the page count for fetching market tokens on mobile platforms from 1 to 2, and increased it for other platforms from 3 to 5 for better data retrieval.

limichange committed:
refactor: adjust layout properties in HolderItemSmall and MobileInformationTabs for improved UI consistency

- Updated width properties for percentage and value in HolderItemSmall to enhance layout responsiveness.
- Added padding to Stack components in MobileInformationTabs for better spacing and visual alignment.

limichange committed:
chore: remove MarketViewToggle component and its related exports

- Deleted MarketViewToggle component and its associated types from the codebase as part of a cleanup effort.

limichange committed:
feat: enhance MobileLayout with icon support for watchlist tab

- Added an Icon component to the watchlist tab in MobileLayout for improved visual representation.
- Updated the name prop handling in Tabs.TabBarItem to conditionally render the icon or the translated text for better user experience.

limichange committed:
refactor: remove ToggleButton component from DesktopLayout for cleaner code

- Deleted the ToggleButton component and its associated props from DesktopLayout to streamline the codebase.
- Adjusted the layout to improve organization and maintainability, focusing on the watchlist tab functionality.

limichange committed:
refactor: update tab names in DesktopLayout for localization support

- Replaced hardcoded tab names with localized strings for the watchlist and trending tabs in DesktopLayout.
- Wrapped the content of the trending tab in a Tabs.ScrollView for improved scrolling behavior and layout consistency.

limichange committed:
Merge branch 'x' into market-update-11

originalix commented on packages/kit/src/views/Market/MarketHomeV2/components/MarketTokenList/hooks/useToDetailPage.ts:
```diff
@@ -13,17 +19,29 @@ interface IMarketToken {
 }
 
 export function useToDetailPage() {
+  const navigation2 = useAppNavigation();
```
appNavigation
